### PR TITLE
fix: JS toggle issue with Numeric Input & multi select dropdown controls

### LINKED ISSUE #14810: [Bug]: File Picker - JS Toggle on Allowed File Types - JS toggle un-clickable till field is cleared
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

With File Picker widget, if we do a JS toggle on Allowed File Types property, with any of the properties selected, the JS button remains non clickable until  the field is cleared. This behavior is seen even if the Allowed file types are of valid file types.

Expected Behavior - If the allowed file types are of valid file types, the JS button should not be disabled. 

### Steps To Reproduce

1. Drop a file picker widget onto the canvas.
2. From the Allowed File Types property, select Images. 
3. Now click on the JS toggle button
4. The button gets disabled immediately and unless we clear the field, it does not get enabled again
5. This behavior is seen even if the Allowed file types are of valid file types, which is incorrect.

If the allowed file types entered are of valid file types, the toggle should not get disabled. Only if we enter some unaccepted value, it should get disabled.
Screenshot 2022-06-24 at 10 14 46 PM

### Public Sample App

_No response_

### Version

Release

### LINKED ISSUE #14867: [Bug]: JS toggle for Label width property is not working properly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Right after clicking the JS button, it becomes disabled.
https://www.loom.com/share/05f6e6c4d8b145398dfe72af6f1c2fdf

### Steps To Reproduce

1. Add any widget with a label width property (example: Input widget)
2. click on JS toggle button for the label width

### Public Sample App

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

NumericInput control which is used for the Label width property was not having a JS toggle validation. Also, the validation logic written for Dropdown control was not handling multi-select. These two validations were added to fix the issues with these controls.

Fixes #14867 
Fixes #14810 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Jest tests are added

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/propertyControls/DropDownControl.tsx [hunk 1]
```diff
@@ -129,14 +129,28 @@ class DropDownControl extends BaseControl<DropDownControlProps> {
...
+  ): boolean {
+    const allowedValues = new Set(
+      config?.options?.map((x: { value: string | number }) =>
+        x.value.toString(),
+      ),
+    );
+    if (config.isMultiSelect) {
+      try {
+        const values = JSON.parse(value);
+        for (const x of values.map((x: string | number) => x.toString())) {
```
Comment (rahulramesha) [inline app/client/src/components/propertyControls/DropDownControl.tsx:144]: Looks like you are just iterating over the list twice first with `.map` and then with `for` loop. Can you just use `x.toString()` below in the if loop?

Also don't feel like a good idea to use `.map` to get the array you are iterating over inside the `for` condition. You could just create the array before and then use that in the `for` condition.

**Review by rahulramesha — CHANGES_REQUESTED** [review]

File: app/client/src/components/propertyControls/DropDownControl.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/components/propertyControls/DropDownControl.tsx:144]: My bad 🤦 

@rahulramesha 
Totally agree with your first point. I will make the change and commit.

But, I didn't get the second point. Did you mention that in terms of readability?

**Review by rahulramesha — APPROVED** [review]

File: app/client/src/components/propertyControls/DropDownControl.tsx (hunk 1 above)
Comment (rahulramesha) [reply app/client/src/components/propertyControls/DropDownControl.tsx]: @aswathkk yes, it is better to not have logic inside conditional statements

**Pull request merged**